docling_serve/engines/async_local/orchestrator.py

```diff
@@ -0,0 +1,101 @@
+import asyncio
+import logging
+import uuid
+from typing import Dict, List, Optional, Set
+
+from fastapi import WebSocket
+
+from docling_serve.datamodel.engines import Task, TaskStatus
+from docling_serve.datamodel.requests import ConvertDocumentsRequest
+from docling_serve.datamodel.responses import (
+    MessageKind,
+    TaskStatusResponse,
+    WebsocketMessage,
+)
+from docling_serve.engines.async_local.worker import AsyncLocalWorker
+from docling_serve.engines.base_orchestrator import BaseOrchestrator
+from docling_serve.settings import docling_serve_settings
+
+_log = logging.getLogger(__name__)
+
+
+class OrchestratorError(Exception):
+    pass
+
+
+class TaskNotFoundError(OrchestratorError):
+    pass
+
+
+class AsyncLocalOrchestrator(BaseOrchestrator):
+    def __init__(self):
+        self.task_queue = asyncio.Queue()
+        self.tasks: Dict[str, Task] = {}
+        self.queue_list: List[str] = []
```

Comment:
Maybe one could encapsulate these three members in a class that derives from asyncio.Queue to make it cleaner and less error prone to use.

Reply:
I think this and other refactoring will become more obvious once we implement the other compute engines (redis, etc)
The idea is also to have functionalities in some base class or objects.